### PR TITLE
Add Notes at Engagement level and improvements at Test level

### PR DESCRIPTION
**Pull request by Maffooch**
Add Notes at the Engagement level as well as include the TypedNotes feature at the test level. Notes at the Engagement level can be found in the Additional Features tab. Requested by @mtesauro 

- [x] Your code is flake8 compliant.
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted).
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR.

### CODE REVIEW

**Comment by valentijnscholten** [comment]
I know this has been open for a while, but still think it would be good to make sure out test suite covers these. At least the basic stuff like adding a note, making sure it's there and displayed OK?

File: dojo/templates/dojo/view_eng.html [hunk 1]
```diff
@@ -514,6 +514,104 @@ <h4 class="modal-title" id="shareSurveyModalLabel">Share Questionnaire Link</h4>
...
+            <i class="glyphicon glyphicon-chevron-down"></i></a></span>
+        </h4>
+      </div>
+      
+      <div id="add_feat_notes" class="panel-body collapse">
+        <form class="form-horizontal" action="" method="post">{% csrf_token %}
+          {% include "dojo/form_fields.html" with form=form %}
+          <div class="form-group">
+              <div class="col-sm-offset-2 col-sm-10">
+                  <input class="btn btn-primary" type="submit" id="add_note" label="Add Notes" value="Add Note"/>
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/templates/dojo/view_eng.html:531]: Label is clearly there

**Comment by valentijnscholten** [comment]
Graet to see test coverage extended. I think it could also be fairly easily added to the api tests in `test_rest_framework.py`?

**Comment by Maffooch** [comment]
Yes I thought so too. Turns out it is not as easy as originally thought. Put method for notes requires more than one arg and returns OrderedDicts. Really struggling to get the update test to pass.

**Comment by Maffooch** [comment]
Boy is it gonna be satisfying to squash all these dusty commits

**Review by valentijnscholten — APPROVED** [review]

**Review by devGregA — APPROVED** [review]

**Pull request merged**